### ISSUE TITLE
[Q/BUG] What is this code supposed to do ?

### ISSUE THREAD

**Issue by qchateau** [Aloxaf/fzf-tab#171]
Everytime I accept or reject a completion, or when I press tab and there is no completion possible, zsh inserts a newline. This is a bit annoying to me. Is this also the case for you or is it related to my zsh config / other plugins ?

The actual question is about the diff below, I noticed when I remove a bit of code (see patch), the problem goes away, and I did not notice any other difference. 
I have 0 knowledge of zsh plugins, I'm just a user, and I don't understand what the code does, but it seems weird that I'm just able to remove code, and it all works as before (and even better in my case).

In short: what did I just do by removing these line ?

```diff
--- a/fzf-tab.zsh
+++ b/fzf-tab.zsh
@@ -195,13 +195,15 @@ fzf-tab-complete() {
     } always {
       IN_FZF_TAB=0
     }
-    if (( _ftb_continue )); then
-      zle .split-undo
-      zle .reset-prompt
-      zle -R
-    else
-      zle redisplay
-    fi
   done
 }
```

**Comment by Aloxaf** [comment]
Sorry for the late reply.

This code is for refreshing the prompt so that so something like zsh-autosuggestions and zsh-syntax-highlighting works. I haven't met this problem, can you provide your zshrc?

**Comment by qchateau** [comment]
Oh I see, then it's probably related to the fact I use the p10k theme with the "transient prompt" feature (and a 2-line prompt btw).

I can post my zshrc+p10k config maybe later (I don't have access to it atm)

**Comment by qchateau** [comment]
Ok I found that when I add the `zsh-autosuggestions` plugin, it suddenly works as expected. Fun. I guess I'll start to use it 8)

FInd both my zshrc and p10k config file below if you're interested
[cfg.zip](https://github.com/Aloxaf/fzf-tab/files/5792552/cfg.zip)

**Comment by Aloxaf** [comment]
I make some improvements in https://github.com/Aloxaf/fzf-tab/commit/db3616cea8f954b73596ccadcdf656484a4fc37b. Please update and try again.

BTW, you should load fzf-tab before zsh-syntax-highlighting & zsh-autosuggestions.

**Comment by qchateau** [comment]
> BTW, you should load fzf-tab before zsh-syntax-highlighting & zsh-autosuggestions.
Done !

I've tried your patch, the problem is still there if I don't have zsh-autosuggestions

Though removing `zle .redisplay` does seem to fix my problem with no side-effect with my config :/

**Comment by Aloxaf** [comment]
Remove `zle .redisplay` in https://github.com/Aloxaf/fzf-tab/commit/209965d98d98256a8081bd8b329c544546d346c7
